pypif_sdk/interop/mdf.py

```diff
@@ -0,0 +1,165 @@
+from pypif_sdk.readview import ReadView
+from pypif.obj.common import Value, ProcessStep
+from citrination_client import PifQuery
+from ..util.citrination import get_client
+from pypif.pif import dumps
+
+import re
+
+
+def query_to_mdf_records(query=None, dataset_id=None):
+    """Evaluate a query and return a list of MDF records
+
+    If a datasetID is specified by there is no query, a simple
+    whole dataset query is formed for the user
+    """
+    if not query and not dataset_id:
+        raise ValueError("Either query or datasetID must be specified")
+    if not query:
+        query = PifQuery(include_datasets=[dataset_id])
+    client = get_client()
+    if not client:
+        raise ValueError("Unable to create a citrination client; is 'CITRINATION_API_KEY' in the env?")
+
+    result = client.search(query)
+    records = []
+    for hit in result.hits:
+        records.append(pif_to_mdf_record(hit.system, hit.dataset))
+
+    return records
+
+
+def pif_to_mdf_record(pif_obj, dataset_id):
+    """Convert a PIF into partial MDF record"""
+    res = {}
+    res["mdf"] = _to_meta_data(pif_obj, dataset_id)
+    res[res["mdf"]["source_name"]] = _to_user_defined(pif_obj)
+    return dumps(res)
+
+
+def _to_meta_data(pif_obj, dataset_id):
+    """Convert the meta-data from the PIF into MDF"""
+    pif = pif_obj.as_dictionary()
+    mdf = {}
+    try:
+        if pif.get("names"):
+            mdf["title"] = pif["names"][0]
+        else:
+            mdf["title"] = "Citrine PIF " + str(pif["uid"])
+        if pif.get("chemicalFormula"):
+            mdf["composition"] = pif["chemicalFormula"]
```

Comment:
Composition information can also be in the `composition` field.  In some cases, it is there but not in `chemicalFormula`, e.g. [here](https://citrination.com/datasets/153258/version/1/pif/242264BBCD27BB025597D8DB7F082CF8)